lib/auth_app_web/user_auth.ex

```diff
@@ -164,10 +167,26 @@ defmodule AuthAppWeb.UserAuth do
     end
   end
 
+  def on_mount(:ensure_sudo_mode, _params, session, socket) do
+    socket = mount_current_user(socket, session)
+
+    if Accounts.sudo_mode?(socket.assigns.current_user, -10) do
+      {:cont, socket}
+    else
+      socket =
+        socket
+        |> Phoenix.LiveView.put_flash(:error, "You must re-authenticate to access this page.")
+        |> Phoenix.LiveView.redirect(to: ~p"/users/log-in")
+
+      {:halt, socket}
+    end
+  end
+
   def on_mount(:redirect_if_user_is_authenticated, _params, session, socket) do
     socket = mount_current_user(socket, session)
+    user = socket.assigns.current_user
 
-    if socket.assigns.current_user do
+    if user && Accounts.sudo_mode?(user, -10) do
```

Comment:
My opinion is to just kill this `on_mount` hook altogether, simplifying the router.

Reply:
so we don't care and just always allow accessing the login page?

Reply:
Yeah, given sudo mode would allow it anyway. We just render it differently we are already signed in? That may require us adding mount_current_user back though.

Reply:
But do we really want to allow visiting the register page when authenticated?

Reply:
Probably not! Maybe we redirect this in the LiveView though? No need for a new router session and the like?

Reply:
https://github.com/SteffenDE/phoenix_gen_auth_magic_link/pull/1/commits/56b0ba69a852dec31bbdae00ffc40c8300900ea7